### PR TITLE
Functional tests - Add new test 'BO > Catalog > Cart rules carrier restriction'

### PR DESCRIPTION
**Pull request by cfarhani06**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |  8.0.x 
| Description?      | BO - Cart rules - Case 11 - Carrier restriction
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+const CartRuleFaker = require('@data/faker/cartRule');
+const ProductData = require('@data/FO/product');
+const {DefaultCustomer} = require('@data/demo/customer');
+const {Products} = require('@data/demo/products');
+const {Carriers} = require('@data/demo/carriers');
+
+// import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_catalog_discounts_cartRules_CRUDCartRule_checkTotalAvailable';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
const baseContext = 'functional_BO_catalog_discounts_cartRules_CRUDCartRule_carrierRestriction';
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  before(async function () {
+    browserContext = await helper.createBrowserContext(this.browser);
+    page = await helper.newTab(browserContext);
+  });
+
+  after(async () => {
+    await helper.closeBrowserContext(browserContext);
+  });
+
+  describe('Create A Cart Rules', async () => {
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
  describe('Create a Cart Rule', async () => {
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 1 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: DONE

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 2 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: DONE

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+      page = await addCartRulePage.viewMyShop(page);
+
+      await foHomePage.changeLanguage(page, 'en');
+      const isHomePage = await foHomePage.isHomePage(page);
+      await expect(isHomePage, 'Fail to open FO home page').to.be.true;
+    });
+  });
+
+  const tests = [
+    {args: {testIdentifier: 'cartRuleAccepted', testTitle: 'for the first time'}},
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Do you have another missed argument ?
Maybe you should remove the foreach

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+          baseContext,
+        );
+
+        await foProductPage.addProductToTheCart(page);
+
+        const notificationsNumber = await cartPage.getCartNotificationsNumber(page);
+        await expect(notificationsNumber).to.be.equal(1);
+      });
+
+      it('should get "You must choose a carrier before applying this voucher" error message',
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Here you add a promo code and you don't get "You must choose a carrier before applying this voucher" error message

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+
+      it('should choose the first shipping method', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          `${test.args.testIdentifier}ShippingMethodStep`,
+          baseContext,
+        );
+
+        await checkoutPage.chooseShippingMethodWithoutValidation(page, 1);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Please add an expect here

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+      it('should choose the second shipping method', async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          `${test.args.testIdentifier}ShippingMethodStep`,
+          baseContext,
+        );
+
+        await checkoutPage.chooseShippingMethodAndAddComment(page, 2);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Please add an expect here

File: tests/UI/data/faker/cartRule.js [hunk 7]
```diff
@@ -14,6 +14,12 @@ class CartRuleData {
    * @param cartRuleToCreate {Object} Could be used to force the value of some members
    */
   constructor(cartRuleToCreate = {}) {
+<<<<<<< HEAD
```
Comment (nesrineabdmouleh) [inline tests/UI/data/faker/cartRule.js]: Please delete it

File: tests/UI/pages/BO/catalog/discounts/add.js [hunk 8]
```diff
@@ -58,6 +58,26 @@ class AddCartRule extends BOBasePage {
     this.quantityInput = 'input[name=quantity]';
     this.quantityPerUserInput = 'input[name=quantity_per_user]';
 
+<<<<<<< HEAD
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/catalog/discounts/add.js]: Please verify the rebase

File: tests/UI/utils/globals.js [hunk 9]
```diff
@@ -40,7 +40,7 @@ global.BROWSER = {
   config: {
     headless: JSON.parse(process.env.HEADLESS || true),
     timeout: 0,
-    slowMo: parseInt(process.env.SLOW_MO, 10) || 5,
+    slowMo: parseInt(process.env.SLOW_MO, 10) || 50,
```
Comment (nesrineabdmouleh) [inline tests/UI/utils/globals.js]: why do you change the slowmo ?

File: tests/UI/pages/FO/checkout/index.js [hunk 10]
```diff
@@ -27,7 +27,27 @@ class Checkout extends FOBasePage {
     this.shippingValueSpan = '#cart-subtotal-shipping span.value';
     this.noPaymentNeededElement = `${this.paymentStepSection} div.content > p.cart-payment-step-not-needed-info`;
     this.noPaymentNeededText = 'No payment needed for this order';
-
+<<<<<<< HEAD
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/FO/checkout/index.js]: Same here

**Comment by nesrineabdmouleh** [comment]
Hello @cfarhani06,
Thank you for your PR!
Please fix reviews and lint errors

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 3 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/data/faker/cartRule.js (hunk 7 above)
Comment (cfarhani06) [reply tests/UI/data/faker/cartRule.js]: done

File: tests/UI/utils/globals.js (hunk 9 above)
Comment (cfarhani06) [reply tests/UI/utils/globals.js]: DONE

File: tests/UI/pages/BO/catalog/discounts/add.js (hunk 8 above)
Comment (cfarhani06) [reply tests/UI/pages/BO/catalog/discounts/add.js]: DONE

File: tests/UI/pages/FO/checkout/index.js (hunk 10 above)
Comment (cfarhani06) [reply tests/UI/pages/FO/checkout/index.js]: DONE

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 4 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: DONE

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 5 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: DONE

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 6 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: DONE

**Comment by cfarhani06** [comment]
> Hello @cfarhani06, Thank you for your PR! Please fix reviews and lint errors

It's OK, you can review !!

File: tests/UI/pages/BO/catalog/discounts/add.js [hunk 11]
```diff
@@ -244,6 +259,7 @@ class AddCartRule extends BOBasePage {
    * Create/edit cart rule
    * @param page {Page} Browser tab
    * @param cartRuleData {CartRuleData} Data to set on add/edit cart rule form
+   * @param checkCarrierRestriction {boolean} True if we need to select the carrier restriction
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/discounts/add.js]:
```suggestion
```

File: tests/UI/data/faker/cartRule.js [hunk 12]
```diff
@@ -71,6 +71,9 @@ class CartRuleData {
     /** @type {number} Amount of times a user can use the cart rule */
     this.quantityPerUser = cartRuleToCreate.quantityPerUser || 1;
 
+    this.countrySelection = cartRuleToCreate.countrySelection || false;
+    this.carrierRestriction = cartRuleToCreate.carrierRestriction || false;
```
Comment (Progi1984) [inline tests/UI/data/faker/cartRule.js]: Please add JsDoc

File: tests/UI/pages/BO/catalog/discounts/add.js (hunk 11 above)
Comment (cfarhani06) [reply tests/UI/pages/BO/catalog/discounts/add.js]: Done

File: tests/UI/data/faker/cartRule.js (hunk 12 above)
Comment (cfarhani06) [reply tests/UI/data/faker/cartRule.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 13]
```diff
@@ -0,0 +1,308 @@
...
+      const notificationsNumber = await cartPage.getCartNotificationsNumber(page);
+      await expect(notificationsNumber).to.be.equal(1);
+    });
+
+    it('should add the promo code and get "You must choose a carrier before applying this voucher" error message',
+      async function () {
+        await testContext.addContextItem(
+          this,
+          'testIdentifier',
+          'AddPromoCode',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
          'addPromoCodeAndGetErrorMessage',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 14]
```diff
@@ -0,0 +1,308 @@
...
+
+        const alertMessage = await cartPage.getCartRuleErrorMessage(page);
+        await expect(alertMessage).to.equal(cartPage.cartRuleAlertMessagetext);
+      });
+
+    it('should proceed to checkouts', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'ProceedToCheckoutAndSignIn',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'proceedToCheckoutAndSignIn',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 15]
```diff
@@ -0,0 +1,308 @@
...
+
+      const pageTitle = await foProductPage.getPageTitle(page);
+      await expect(pageTitle.toUpperCase()).to.contains(ProductData.firstProductData.name);
+    });
+
+    it('should add product to cart and proceed to checkout', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'AddProductToCart',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'addProductToCart',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 16]
```diff
@@ -0,0 +1,308 @@
...
+      await expect(isHomePage, 'Fail to open FO home page').to.be.true;
+    });
+  });
+
+  describe('Use Cart Rule', async () => {
+    it('should go to the first product page', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'GoToFirstProductPage',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'goToFirstProductPage',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 17]
```diff
@@ -0,0 +1,308 @@
...
+          this,
+          'testIdentifier',
+          'AddPromoCode',
+          baseContext,
+        );
+
+        await cartPage.addPromoCode(page, cartRuleCode.code);
+
+        const alertMessage = await cartPage.getCartRuleErrorMessage(page);
+        await expect(alertMessage).to.equal(cartPage.cartRuleAlertMessagetext);
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        await expect(alertMessage).to.equal(cartPage.cartRuleAlertMessageText);
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 18]
```diff
@@ -0,0 +1,308 @@
...
+
+      const isCheckout = await checkoutPage.isCheckoutPage(page);
+      await expect(isCheckout).to.be.true;
+    });
+
+    it('should be authentificated', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'SignInFO',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'signInFO',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 19]
```diff
@@ -0,0 +1,308 @@
...
+
+      const isCustomerConnected = await checkoutPage.customerLogin(page, DefaultCustomer);
+      await expect(isCustomerConnected, 'Customer is not connected').to.be.true;
+    });
+
+    it('should confirm adress after signIn', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'ConfirmAdressStep',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'confirmAddressStep',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 20]
```diff
@@ -0,0 +1,308 @@
...
+        'testIdentifier',
+        'addPromoCode',
+        baseContext,
+      );
+
+      await checkoutPage.chooseShippingMethodAndAddComment(page, 1);
+      await checkoutPage.addPromoCode(page, cartRuleCode.code);
+
+      const errorShippingMessage = await checkoutPage.getCartRuleErrorMessage(page);
+      await expect(errorShippingMessage).to.equal(checkoutPage.cartRuleAlertMessagetext);
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
      await expect(errorShippingMessage).to.equal(checkoutPage.cartRuleAlertMessageText);
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 21]
```diff
@@ -0,0 +1,308 @@
...
+
+      const isDeliveryStep = await checkoutPage.goToDeliveryStep(page);
+      await expect(isDeliveryStep, 'Delivery Step boc is not displayed').to.be.true;
+    });
+
+    it('should set the promo code and choose the wrong shipping method', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'addPromoCode',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'addPromoCodeAndChooseWrongShippingMethod',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 22]
```diff
@@ -0,0 +1,308 @@
...
+
+      const errorShippingMessage = await checkoutPage.getCartRuleErrorMessage(page);
+      await expect(errorShippingMessage).to.equal(checkoutPage.cartRuleAlertMessagetext);
+    });
+
+    it('should set the promo code for second time and check total after discount', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'AddPromoCodeANDVerifyTotalAfterDiscount',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'addPromoCodeANDVerifyTotalAfterDiscount',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 23]
```diff
@@ -0,0 +1,308 @@
...
+          .toFixed(2),
+      ),
+      );
+    });
+
+    it('should remove the discount', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'RemoveTheDiscount',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'removeTheDiscount',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 24]
```diff
@@ -0,0 +1,308 @@
...
+
+      const isDeleteIconNotVisible = await checkoutPage.removePromoCode(page);
+      await expect(isDeleteIconNotVisible, 'The discount is not removed').to.be.true;
+    });
+
+    it('should click on the logo of the shop', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'CheckLogoLink',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'checkLogoLink',
```

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js [hunk 25]
```diff
@@ -0,0 +1,308 @@
...
+      await foHomePage.clickOnHeaderLink(page, 'Logo');
+      const pageTitle = await foHomePage.getPageTitle(page);
+      await expect(pageTitle).to.equal(foHomePage.pageTitle);
+    });
+
+    it('should click on the cart link', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'ClickOnCartLink',
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]:
```suggestion
        'clickOnCartLink',
```

**Review by Progi1984 — COMMENTED** [review]
Small fixes (identifiers are in camelCase)

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 13 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 14 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 15 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 16 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 17 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 18 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 19 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 20 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 21 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 22 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 25 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 24 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js (hunk 23 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js]: Done

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
thanks @cfarhani06 :)

**Pull request merged**